src/log_impl.rs

```diff
@@ -116,6 +120,40 @@ pub struct Panic;
 
 pub struct Null;
 
+//Custom file logger which we will be able to manipulate the name
+#[derive(Debug)]
+pub struct CustomFileLog{
+    pub line_sep:  Cow<'static, str>,
+    pub file_suffix: String,
+    pub file_name: String,
+    pub curr_file_suffix: Mutex<String>,
+    pub is_writable: Mutex<bool>,
+    pub file_stream: Mutex<BufWriter<fs::File>>,
```

Comment:
Since we only ever access these three mutexes one after another, what would you think of putting all the data inside the same mutex?

We could have an inner struct like

```
struct CustomFileLogState {
    curr_file_suffix: String,
    is_writeable: bool,
    file_stream: BufWriter<fs::File>,
}
```

Then just store `Mutex<CustomFileLogInner>` in `CustomFileLog`. The main advantage would be speed: we would avoiding locking two or three separate mutexes in sequence when we could instead be locking a single one.